src/utils/address.ts

```diff
@@ -21,8 +22,80 @@ export function toChecksum(address) {
   return utils.getAddress(address);
 }
 
-export function toAddressStub(address) {
-  const start = address.slice(0, 5);
-  const end = address.slice(-3);
-  return `${start}...${end}`;
+export function toAddressStub(address, size = 'default') {
+  const start = address.slice(0, 6);
+  const end = address.slice(-4);
+
+  switch (size) {
+    case 'short':
+      return `${start}..`;
+    case 'long':
+      return address;
+    default:
+      return `${start}...${end}`;
+  }
+}
+
+export function getBlockchainLink(address, networkName, type) {
+  switch (type) {
+    case 'user':
+      return `${window.location.pathname}#/user/${address}`;
+    case 'address':
+      if (networkName === 'arbitrum')
+        return `https://explorer.arbitrum.io/#/address/${address}`;
+      else if (networkName === 'arbitrumTestnet')
+        return `https://rinkeby-explorer.arbitrum.io/#/address/${address}`;
+      else if (networkName === 'mainnet')
+        return `https://etherscan.io/address/${address}`;
+      else if (networkName === 'xdai')
+        return `https://blockscout.com/xdai/mainnet/address/${address}`;
+      else return `https://${networkName}.etherscan.io/address/${address}`;
+    default: // investigate DRY here
+      if (networkName === 'arbitrum')
+        return `https://explorer.arbitrum.io/#/tx/${address}`;
+      else if (networkName === 'arbitrumTestnet')
+        return `https://rinkeby-explorer.arbitrum.io/#/tx/${address}`;
+      else if (networkName === 'mainnet')
+        return `https://etherscan.io/tx/${address}`;
+      else if (networkName === 'xdai')
+        return `https://blockscout.com/xdai/mainnet/tx/${address}`;
+      else return `https://${networkName}.etherscan.io/tx/${address}`;
+  }
+}
+
+export async function getENSName(address) {
+  let name = null;
+  try {
+    const provider = ethers.getDefaultProvider();
+    let checksumed = ethers.utils.getAddress(address);

```

Comment:
I don't understand why this is a variable and not just a constant?.

Reply:
uf, bad on me.